zbus/src/object_server/mod.rs

```diff
@@ -782,8 +782,35 @@ impl ObjectServer {
             })?
         };
 
-        self.dispatch_call_from_iface(iface, connection, msg, hdr)
-            .await
+        trace!("acquiring read lock on interface `{}`", iface_name);
+        let read_lock = iface.read().await;
```

Comment:
* `read_lock` here is not a lock but a guard.
* I prefer naming the guards by what they contain (rather than naming it after their smart pointer container), so:

```rust
let with_spawn = {
     ...
     let iface = iface.read().await;
     ...
     iface.with_spawn()
};
```

This also gets rid of explicit `drop`. Call me a weirdo but I prefer using scopes over manual drops. :)

Reply:
This is based on existing code, I just followed the same style. The dispatch usage requires different variables with explicit drop because of the read/write locking.

It can't be done the way you suggest.

Reply:
> The dispatch usage requires different variables with explicit drop because of the read/write locking.

Not sure, I follow. why can't my suggestion work? scoping is a very usual way to drop locks as soon as they're no longer needed.

Reply:
> This is based on existing code, I just followed the same style.

Pretty sure, the existing code doesn't call this boolean returning method, since you added it in this PR. :)

Reply:
ok, I added a scope to drop the explicit drop